manual_scans/aws/aws_alias_cloudfront_s3.py

```diff
@@ -4,22 +4,25 @@
 import boto3
 import requests
 
+from utils.utils_aws_manual import bucket_does_not_exist
+from utils.utils_aws_manual import get_cloudfront_origin_url
+from utils.utils_aws_manual import is_s3_bucket_url
+from utils.utils_aws_manual import is_s3_website_endpoint_url
 from utils.utils_aws_manual import list_hosted_zones_manual_scan
 from utils.utils_print import my_print
 from utils.utils_print import print_list
 
 
-vulnerable_domains = []
-missing_resources = []
-
-
 def vulnerable_alias_cloudfront_s3(domain_name):
-
     try:
         response = requests.get(f"https://{domain_name}", timeout=1)
 
-        if response.status_code == 404 and "Code: NoSuchBucket" in response.text:
-            return True
+        if response.status_code == 404 and "<Code>NotFound</Code>" in response.text:
+            bucket_url = get_cloudfront_origin_url(domain_name)
+            if not is_s3_bucket_url(bucket_url) and not is_s3_website_endpoint_url(bucket_url):
+                return False
+
+            return bucket_does_not_exist(bucket_url)
```

Comment:
Going to a standard S3 bucket URL returns the `NoSuchBucket` code. However when going to the S3 website endpoint for a bucket which doesn't exist, the request seems to just time out. 

To cater for this, I'd suggest normalising the S3 bucket URL before testing for `NoSuchBucket`, I did something similar already [here](https://github.com/domain-protect/domain-protect/blob/0d74d8305e7655130478caa8e0f5532e4bc77cba/lambda_code/takeover/takeover.py#L351) though looking at this now I can see it should also include the `.s3-website.` syntax for completeness.

Reply:
> when going to the S3 website endpoint for a bucket which doesn't exist, the request seems to just time out

This doesn't seem to be the case for me:

```
$ curl idontexistatall.s3-website-us-east-1.amazonaws.com
<html>
<head><title>404 Not Found</title></head>
<body>
<h1>404 Not Found</h1>
<ul>
<li>Code: NoSuchBucket</li>
<li>Message: The specified bucket does not exist</li>
<li>BucketName: idontexistatall</li>
<li>RequestId: WW4DWGDQAH4GCXTV</li>
<li>HostId: EOJUtUEw4ufqSa8sAw9S+pXaXx1xGwhf+0lw9ZwLdFavnIIa0ksmSimMvqYufJINaU2w5iYP0bI=</li>
</ul>
<hr/>
</body>
</html>
```

Same for the dot notation e.g.

```
$ curl idontexistatall.s3-website.ap-south-1.amazonaws.com
<html>
<head><title>404 Not Found</title></head>
<body>
<h1>404 Not Found</h1>
<ul>
<li>Code: NoSuchBucket</li>
<li>Message: The specified bucket does not exist</li>
<li>BucketName: idontexistatall</li>
<li>RequestId: GBTDCV60Q766TXX3</li>
<li>HostId: FpMTYDjlPhTxomlu5K4ah5SScuKJYBtF0ic47WmnpZtZqhSWtUH45mQdLUFBaKyGfyC1sxisJc8=</li>
</ul>
<hr/>
</body>
</html>
```

Reply:
That said, the code was not detecting bucket not found properly for these. Fixed in [c1a8222](https://github.com/domain-protect/domain-protect/pull/246/commits/c1a82222943ad580879d7ba25db9ef0622712909)

Reply:
seems to be working for me now